server/channels/store/sqlstore/preference_store.go

```diff
@@ -315,3 +315,52 @@ func (s SqlPreferenceStore) CleanupFlagsBatch(limit int64) (int64, error) {
 
 	return rowsAffected, nil
 }
+
+func (s SqlPreferenceStore) DeleteInvalidVisibleDmsGms() (int64, error) {
+	var query string
+	if s.DriverName() == "postgres" {
+		query = `DELETE FROM Preferences
+		WHERE (userid, category, name) IN (
+			SELECT userid, category, name 
+			FROM preferences 
+			WHERE category = 'sidebar_settings' 
+			AND name = 'limit_visible_dms_gms'
+			AND (SUBSTRING(
+			  CONCAT('000000000000000', value), 
+			  LENGTH(value) + 1, 
+			  15
+			) > '000000000000040'
+			OR SUBSTRING(
+				CONCAT('000000000000000', value), 
+				LENGTH(value) + 1, 
+				15
+			  ) < '000000000000001'
+			) 
```

Comment:
This is very hard to read. And there are no comments, which isn't making things easier. And everything is hardcoded with no parameterization.

1. I would suggest to use squirrel to build up the query. Use constants like `model.PreferenceCategorySidebarSettings` instead of hardcoded strings and pass them as parameters in the query.
2. Add some comments to explain what is being achieved here.

Reply:
I've changed this to use squirrel and added comments for clarifications.